classes/Cart.php

```diff
@@ -4827,6 +4827,23 @@ public static function isGuestCartByCartId($id_cart)
             WHERE ca.`id_cart` = ' . (int) $id_cart);
     }
 
+    /**
+     * Are all products of the Cart still exist?
+     *
+     * @return bool False if not all products in the cart still exist
+     */
+    public function isAllProductsStillExist()
```

Comment:
```suggestion
    public function checkAllProductsAreStillAvailableInThisState()
```
If I get it right, this function checks that the products in the cart have not been transformed from a standard product to a product-with-combinations 😉so we dont check they "still exist" but rather they "still are available in this state"

Reply:
Maybe something a bit shorter? like `checkProductsTypeChanged` ?

Reply:
commit with @matks version